### PR TITLE
Add current_forecast, today_forecast, nearest_forecast

### PR DESCRIPTION
**Pull request by oncleben31**

### CODE REVIEW

File: src/meteofrance/model/forecast.py [hunk 1]
```diff
@@ -38,6 +38,31 @@ def probability_forecast(self) -> list:
         """Return the wheather event forecast."""
         return self.raw_data["probability_forecast"]
 
+    @property
+    def today_forecast(self) -> dict:
+        """Return the forecast for today."""
+        return self.raw_data["daily_forecast"][0]
```
Comment (Quentame) [inline src/meteofrance/model/forecast.py:44]:
```suggestion
        return self.daily_forecast[0]
```

File: src/meteofrance/model/forecast.py [hunk 2]
```diff
@@ -38,6 +38,31 @@ def probability_forecast(self) -> list:
...
+
+    @property
+    def current_forecast(self) -> dict:
+        """Return the forecast of the current hour."""
+        current_hour_timestamp = (
+            datetime.utcnow().replace(minute=0, second=0, microsecond=0).timestamp()
+        )
+        return next(
+            (x for x in self.forecast if x["dt"] == current_hour_timestamp),
+            self.forecast[0],
```
Comment (Quentame) [inline src/meteofrance/model/forecast.py:63]: If we don't find the right forecast, we should return `None` so we don't confuse to give the wrong data.

```suggestion
            None,
```

File: tests/test_forecast.py [hunk 3]
```diff
@@ -28,6 +30,17 @@ def test_forecast():
         )
         == datetime
     )
+    assert abs(weather_forecast.nearest_forecast["dt"] - now_ts) == min(
+        [abs(x["dt"] - now_ts) for x in weather_forecast.forecast]
+    )
+    assert (
+        current_forecast["dt"] < datetime.utcnow().timestamp()
+        or current_forecast["dt"] == weather_forecast.forecast[0]["dt"]
```
Comment (Quentame) [inline tests/test_forecast.py:38]: With previous comment

```suggestion
        or not current_forecast["dt"]
```

File: tests/test_forecast.py [hunk 4]
```diff
@@ -35,7 +35,7 @@ def test_forecast():
     )
     assert (
         current_forecast["dt"] < datetime.utcnow().timestamp()
-        or current_forecast["dt"] == weather_forecast.forecast[0]["dt"]
+        or not current_forecast["dt"]
```
Comment (Quentame) [inline tests/test_forecast.py:38]:
```suggestion
        or not current_forecast
```

File: src/meteofrance/model/forecast.py [hunk 5]
```diff
@@ -38,6 +38,31 @@ def probability_forecast(self) -> list:
...
+    @property
+    def current_forecast(self) -> dict:
+        """Return the forecast of the current hour."""
+        current_hour_timestamp = (
+            datetime.utcnow().replace(minute=0, second=0, microsecond=0).timestamp()
+        )
+        return next(
+            (x for x in self.forecast if x["dt"] == current_hour_timestamp),
+            None,
+        )
```
Comment (Quentame) [inline src/meteofrance/model/forecast.py:64]: simpler than next, you have:
`forecast_by_datetime ={item["dt"]:item for item in self.forecast}`

Then `return forecast_by_datetime.get(current_hour_timestamp)`

Comment (oncleben31) [reply src/meteofrance/model/forecast.py]: OK taken into account to have a better code coverage.

File: tests/test_forecast.py [hunk 6]
```diff
@@ -13,7 +14,7 @@ def test_forecast():
     client = MeteoFranceClient()
 
     weather_forecast = client.get_forecast(latitude=48.8075, longitude=2.24028)
```
Comment (Quentame) [inline tests/test_forecast.py:16]: When testing with Montréal, Canada:
`weather_forecast = client.get_forecast(latitude=45.5016889, longitude=-73.567256)`

for `current, got`: `KeyError: 1591556400`
for `nearest` got dt: 1591552800

my dt was 1591556947

https://www.epochconverter.com

PS: `probability_forecast` raised `KeyError: 'probability_forecast'`, as it's available only in France. I think we should handle it, returning an empty list can be ok.

Comment (oncleben31) [reply tests/test_forecast.py:16]: Error came from for world cities, forecast don't give a forecast every  hours.
We agreed on an returning `nearest_forecast` when current is not found.

File: src/meteofrance/model/forecast.py [hunk 7]
```diff
@@ -36,7 +36,49 @@ def forecast(self) -> list:
     @property
     def probability_forecast(self) -> list:
         """Return the wheather event forecast."""
-        return self.raw_data["probability_forecast"]
+        if "probability_forecast" in self.raw_data:
+            probability_forecast = self.raw_data["probability_forecast"]
+        else:
+            probability_forecast = []
+        return probability_forecast
```
Comment (Quentame) [inline src/meteofrance/model/forecast.py:43]:
```suggestion
        return self.raw_data.get("probability_forecast", [])
```

File: src/meteofrance/model/forecast.py [hunk 8]
```diff
@@ -36,7 +36,49 @@ def forecast(self) -> list:
...
+        forecast_by_datetime = {item["dt"]: item for item in self.forecast}
+        if current_hour_timestamp in forecast_by_datetime.keys():
+            # Return the element corresponding to the timesamp of the current hour if
+            # exsit.
+            current_forecast = forecast_by_datetime[current_hour_timestamp]
+        else:
+            # Else: return nearest forecast.
+            current_forecast = self.nearest_forecast
+
+        return current_forecast
```
Comment (Quentame) [inline src/meteofrance/model/forecast.py:81]:
```suggestion
        # Return the forecast corresponding to the timestamp of the current hour if exists
        # If not exists, returns the nearest forecast (not France countries)
        return forecast_by_datetime.get(current_hour_timestamp, self.nearest_forecast)
```

File: tests/test_forecast.py [hunk 9]
```diff
@@ -28,6 +31,44 @@ def test_forecast():
         )
         == datetime
     )
+    assert abs(weather_forecast.nearest_forecast["dt"] - now_ts) <= 30 * 60
+    assert now_ts - 3600 <= current_forecast["dt"] <= now_ts
```
Comment (Quentame) [inline tests/test_forecast.py:35]:
```suggestion
    assert now_ts - 3600 <= weather_forecast.current_forecast["dt"] <= now_ts
```

File: tests/test_forecast.py [hunk 10]
```diff
@@ -8,11 +9,13 @@
...
 
 
-def test_forecast():
+def test_forecast_france():
     """Test weather forecast results from API."""
     client = MeteoFranceClient()
 
     weather_forecast = client.get_forecast(latitude=48.8075, longitude=2.24028)
+    now_ts = int(time.time())
+    current_forecast = weather_forecast.current_forecast
```
Comment (Quentame) [inline tests/test_forecast.py:18]:
```suggestion
```

File: tests/test_forecast.py [hunk 11]
```diff
@@ -28,6 +31,44 @@ def test_forecast():
...
+            weather_forecast.timestamp_to_locale_time(
+                weather_forecast.daily_forecast[0]["dt"]
+            )
+        )
+        == datetime
+    )
+    assert abs(weather_forecast.nearest_forecast["dt"] - now_ts) == min(
+        abs(x["dt"] - now_ts) for x in weather_forecast.forecast
+    )
+    assert current_forecast["dt"] == weather_forecast.nearest_forecast["dt"]
```
Comment (Quentame) [inline tests/test_forecast.py:67]:
```suggestion
    assert weather_forecast.current_forecast["dt"] == weather_forecast.nearest_forecast["dt"]
```

File: tests/test_forecast.py [hunk 12]
```diff
@@ -28,6 +31,44 @@ def test_forecast():
...
+    )
+
+
+def test_forecast_world():
+    """Test weather forecast results from API."""
+    client = MeteoFranceClient()
+
+    weather_forecast = client.get_forecast(latitude=45.5016889, longitude=73.567256)
+    now_ts = int(time.time())
+    current_forecast = weather_forecast.current_forecast
```
Comment (Quentame) [inline tests/test_forecast.py:48]:
```suggestion
```

**Comment by oncleben31** [comment]
Nice collaboration. Thanks.

**Pull request merged**